src/main/groovy/jenkins/automation/utils/CheckmarxUtils.groovy

```diff
@@ -0,0 +1,172 @@
+
+    static void checkmarxScan(context, Map cxConfig) {
+        assert cxConfig.projectName != null
+        assert cxConfig.groupId != null
+
+        def useOwnServerCredentialsDefault = false
+        def presetDefault = "17" // "Default 2014"
+        def excludeFoldersDefault = "resources, .git"
+        def incrementalDefault = true
+        def fullScanCycleDefault = "10"
+        def commentDefault = ""
+        def vulnerabilityThresholdDefault = true
+        def highThresholdDefault = 1
+        def mediumThresholdDefault = 2
+        def lowThresholdDefault = 3
+        def filterPatternDefault = """
+            !**/_cvs/**/*, !**/.svn/**/*,   !**/.hg/**/*,   !**/.git/**/*,  !**/.bzr/**/*, !**/bin/**/*,
+            !**/obj/**/*,  !**/backup/**/*, !**/.idea/**/*, !**/*.DS_Store, !**/*.ipr,     !**/*.iws,
+            !**/*.bak,     !**/*.tmp,       !**/*.aac,      !**/*.aif,      !**/*.iff,     !**/*.m3u, !**/*.mid, !**/*.mp3,
+            !**/*.mpa,     !**/*.ra,        !**/*.wav,      !**/*.wma,      !**/*.3g2,     !**/*.3gp, !**/*.asf, !**/*.asx,
+            !**/*.avi,     !**/*.flv,       !**/*.mov,      !**/*.mp4,      !**/*.mpg,     !**/*.rm,  !**/*.swf, !**/*.vob,
+            !**/*.wmv,     !**/*.bmp,       !**/*.gif,      !**/*.jpg,      !**/*.png,     !**/*.psd, !**/*.tif, !**/*.swf,
+            !**/*.jar,     !**/*.zip,       !**/*.rar,      !**/*.exe,      !**/*.dll,     !**/*.pdb, !**/*.7z,  !**/*.gz,
+            !**/*.tar.gz,  !**/*.tar,       !**/*.gz,       !**/*.ahtm,     !**/*.ahtml,   !**/*.fhtml, !**/*.hdm,
+            !**/*.hdml,    !**/*.hsql,      !**/*.ht,       !**/*.hta,      !**/*.htc,     !**/*.htd, !**/*.war, !**/*.ear,
+            !**/*.htmls,   !**/*.ihtml,     !**/*.mht,      !**/*.mhtm,     !**/*.mhtml,   !**/*.ssi, !**/*.stm,
+            !**/*.stml,    !**/*.ttml,      !**/*.txn,      !**/*.xhtm,     !**/*.xhtml,   !**/*.class, !**/*.iml, !Checkmarx/Reports/*.*
+        """.stripIndent()
+
+        context.with {
+            configure {
+                it / builders / 'com.checkmarx.jenkins.CxScanBuilder' {
+
+                    'useOwnServerCredentials'(
+                        cxConfig.get(
+                            "useOwnServerCredentials",
+                            useOwnServerCredentialsDefault
+                        )
+                    )
+
+                    if (cxConfig.useOwnServerCredentials) {
+                        'serverUrl'(cxConfig.serverUrl)
+                        'username'(cxConfig.username)
+                        'password'(cxConfig.password)
+                    }
+
+                    'projectName'(cxConfig.projectName)
+                    'groupId'(cxConfig.groupId) // really the Checkmarx 'team'
+                    'preset'(cxConfig.get("preset", presetDefault))
+
+                    // TODO figure out whether this does anything
+                    'presetSpecified'(false)
+
+                    // TODO probably this should be configurable
```

Comment:
Seems like this functionality might be desirable. https://github.cfpb.gov/HMDA-Operations/hmda-devops/blob/master/jenkins/apps/env.groovy#L108 . Though they're not using this builder.